### PR TITLE
Improve title in void-dom-elements-no-children.md

### PR DESCRIPTION
**Pull request by omerzach**
Add the name of the rule in parentheses at the end of Markdown title like in other documentation files.

### CODE REVIEW

**Comment by ljharb** [comment]
Replaces #1078

File: docs/rules/void-dom-elements-no-children.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Prevent void DOM elements (e.g. `<img />`, `<br />`) from receiving children
+# Prevent void DOM elements (e.g. `<img />`, `<br />`) from receiving children (react/void-dom-elements-no-children)
```
Comment (yannickcr) [inline docs/rules/void-dom-elements-no-children.md]: We did not put the `react/` prefix in the other titles, I think it's not necessary to put it here either

Comment (ljharb) [inline docs/rules/void-dom-elements-no-children.md]: This was my request, per https://github.com/yannickcr/eslint-plugin-react/pull/1078#discussion_r101881565

Comment (yannickcr) [inline docs/rules/void-dom-elements-no-children.md]: Ho, did now saw your comment and #1071. Ok then :)

**Pull request merged**